cmd/cronosd/cmd/root.go

```diff
@@ -220,61 +216,25 @@ type appCreator struct {
 
 // newApp is an AppCreator
 func (a appCreator) newApp(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts servertypes.AppOptions) servertypes.Application {
-	var cache sdk.MultiStorePersistentCache
-
-	if cast.ToBool(appOpts.Get(server.FlagInterBlockCache)) {
-		cache = store.NewCommitKVStoreCacheManager()
+	snapshotDir := filepath.Join(cast.ToString(appOpts.Get(flags.FlagHome)), "data", "snapshots")
+	if err := os.MkdirAll(snapshotDir, os.ModePerm); err != nil {
+		panic(err)
 	}
```

Comment:
```suggestion
```

Reply:
sdk contains this [part](https://github.com/cosmos/cosmos-sdk/blob/v0.46.9-rc.0/server/util.go#L436)

Reply:
no mkdir

Reply:
actually a bit different, [0o755](https://github.com/cosmos/cosmos-sdk/blob/v0.46.9-rc.0/snapshots/store.go#L39) vs ModePerm [0o777](https://pkg.go.dev/os)

Reply:
yeah, we need to create the dir before open the db.